src/preact/base-element.js

```diff
@@ -226,14 +340,19 @@ export class PreactBaseElement extends BaseElement {
   /**
    * A chance to initialize default Preact props for the element.
    *
-   * @return {!JsonObject|undefined}
+   * @return {!JsonObject|void}
    */
   init() {}
 
-  /** @override */
+  /**
+   * @override
+   * @param {import('#core/dom/layout').Layout_Enum} layout
+   */
   isLayoutSupported(layout) {
-    const Ctor = this.constructor;
-    if (Ctor['layoutSizeDefined']) {
+    const Ctor = /** @type {typeof PreactBaseElement} */ (
+      /** @type {unknown} */ (this.constructor)
```

Comment:
nit: consistency with other JSDoc-style any/unknown
```suggestion
      /** @type {?} */ (this.constructor)
```